tests/src/helpers.c

```diff
@@ -282,3 +282,58 @@ void mbedtls_param_failed( const char *failure_condition,
     }
 }
 #endif /* MBEDTLS_CHECK_PARAMS */
+
+#if defined(MBEDTLS_TEST_HOOKS)
+void mbedtls_test_err_add_check( int high, int low,
+                                 const char *file, int line )
+{
+    /* Error codes are always negative (a value of zero is a success) however
+     * their positive opposites can be easier to understand. The following
+     * examples given in comments have been made positive for ease of
+     * understanding. The structure of an error code is such:
+     *
+     *                                                shhhhhhhhlllllll
+     *
+     * s = sign bit.
+     * h = high level error code (includes high and module error codes).
+     * l = low level error code.
+     */
+    if ( high > -0x1000 && high != 0 )
+    /* high < 0001000000000000
+     * No high level error bits are set.
```

Comment:
```suggestion
     * No high level module ID bits are set.
```

Reply:
No I do mean "high level error bits" here. We cannot have an error code with just module ID bits without any high level error bits set.

Reply:
I am referring to the documentation in error.h for the "naming of the bits":
```c 
16 bit error code bit-segmentation                                           
                                                                               
1 bit  - Unused (sign bit)                                                   
3 bits - High level module ID                                                
5 bits - Module-dependent error code                                         
7 bits - Low level module errors
```
I think we are talking about the same thing here a.k.a bits 12..14. It is just that according to the doc copied above the proper name for those bits is "High level module ID".

Reply:
Thanks for the clarification, I missed their precise name in the documentation. Now fixed.